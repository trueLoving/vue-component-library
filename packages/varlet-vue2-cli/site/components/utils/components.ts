import Vue, { ComponentOptions } from 'vue'
import { CombinedVueInstance } from 'vue/types/vue'

export function pickProps<T, U extends keyof T>(props: T, propsKey: U): T[U]
export function pickProps<T, U extends keyof T>(props: T, propsKey: U[]): Pick<T, U>
export function pickProps(props: any, propsKey: any): any {
  return Array.isArray(propsKey)
    ? propsKey.reduce((pickedProps: any, key) => {
        pickedProps[key] = props[key]
        return pickedProps
      }, {})
    : props[propsKey]
}

export interface MountComponentApi {
  instance: CombinedVueInstance<any, any, any, any, any>
  unmount(): void
}

export function mountComponent(component: ComponentOptions<Vue>, container = 'body', options = {}): MountComponentApi {
  const instance = new (Vue.extend(component))(options)
  const el = instance.$mount().$el
  const wrapper = document.querySelector(container)!
  wrapper.appendChild(el)

  return {
    instance,
    unmount() {
      instance.$destroy()
      wrapper.removeChild(el)
    },
  }
}
