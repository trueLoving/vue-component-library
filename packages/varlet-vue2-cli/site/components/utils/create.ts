import Vue, { ComponentOptions, RenderContext, VueConstructor } from 'vue'
import { SlotsMixin } from './mixins/slots'
import { ListenersMixin } from './mixins/listeners'

export type ComponentOptionsPlugin = ComponentOptions<Vue> & { install?: (app: VueConstructor) => void }

export function nextTick() {
  return new Promise(Vue.nextTick)
}

export function mergeSlots(context: RenderContext) {
  const scopedSlots = context.scopedSlots ?? context.data.scopedSlots ?? {}
  const slots = context.slots()

  Object.keys(slots).forEach((key) => {
    if (!scopedSlots[key]) {
      scopedSlots[key] = () => slots[key]
    }
  })

  return scopedSlots
}

export function defineComponent(sfc: ComponentOptionsPlugin & { name: string }) {
  if (sfc.mixins) {
    sfc.mixins.push(SlotsMixin)
    sfc.mixins.push(ListenersMixin)
  } else {
    sfc.mixins = [SlotsMixin, ListenersMixin]
  }

  return sfc
}
