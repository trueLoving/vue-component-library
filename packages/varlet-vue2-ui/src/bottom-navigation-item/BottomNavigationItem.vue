<template>
  <button
    class="var-bottom-navigation-item"
    ref="bottomNavigationItem"
    v-ripple
    :style="{
      color: computeColorStyle(),
    }"
    @click="handleClick"
  >
    <var-icon
      v-if="icon && !$slots.icon"
      :name="icon"
      :namespace="namespace"
      class="var-bottom-navigation-item__icon"
    />
    <slot name="icon" :active="isActive"></slot>
    <var-badge v-if="badge" v-bind="badgeProps" class="var-bottom-navigation-item__badge" />
    <span class="var-bottom-navigation-item__label">
      <template v-if="!$slots.default">
        {{ label }}
      </template>
      <slot></slot>
    </span>
  </button>
</template>

<script>
import Ripple from '../ripple'
import VarBadge from '../badge'
import VarIcon from '../icon'
import { defineComponent } from '../utils/create'
import { props } from './props'
import { createChildrenMixin } from '../utils/mixins/relation'

const defaultBadgeProps = {
  type: 'danger',
  dot: true,
}

export default defineComponent({
  name: 'VarBottomNavigationItem',
  components: {
    VarBadge,
    VarIcon,
  },
  directives: { Ripple },

  mixins: [
    createChildrenMixin('bottomNavigation', { parentKey: 'bottomNavigation', childrenKey: 'bottomNavigationItems' }),
  ],

  inheritAttrs: false,
  props,

  data: () => ({
    isActive: false,
    badgeProps: {},
  }),

  mounted() {
    this.isActive = this.name === this.bottomNavigation.active || this.index === this.bottomNavigation.active

    if (this.isActive) {
      this.$refs.bottomNavigationItem.classList.add('var-bottom-navigation-item--active')
    }
  },

  watch: {
    badge: {
      handler(newValue) {
        this.badgeProps = newValue === true ? defaultBadgeProps : this.badge
      },
      immediate: true,
    },
  },

  methods: {
    computeColorStyle() {
      const { activeColor, inactiveColor } = this.bottomNavigation

      return this.isActive ? activeColor : inactiveColor
    },
    setCurrent(value) {
      const _isActive = value === this.index || value === this.name

      if (_isActive === this.isActive) {
        return
      }

      this.isActive = _isActive

      if (this.isActive) {
        this.$refs.bottomNavigationItem.classList.add('var-bottom-navigation-item--active')
      } else {
        this.$refs.bottomNavigationItem.classList.remove('var-bottom-navigation-item--active')
      }
    },
    handleClick() {
      const { name, getListeners, index } = this
      const active = name ?? index

      getListeners().onClick?.(active)
      this.bottomNavigation.onToggle(active)
    },
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../ripple/ripple';
@import '../badge/badge';
@import '../icon/icon';
@import './bottomNavigationItem';
</style>
