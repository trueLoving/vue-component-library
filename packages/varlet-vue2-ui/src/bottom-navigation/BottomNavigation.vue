<template>
  <div
    class="var-bottom-navigation"
    :class="[fixed ? 'var-bottom-navigation--fixed' : null, border ? 'var-bottom-navigation--border' : null]"
    ref="bottomNavigationDom"
    :style="`z-index:${zIndex}`"
  >
    <slot></slot>

    <var-button
      v-if="$slots.fab"
      class="var-bottom-navigation__fab"
      :class="
        bottomNavigationItems.length % 2 ? 'var-bottom-navigation--fab-right' : 'var-bottom-navigation--fab-center'
      "
      @click="handleFabClick"
      v-bind="_fabProps"
      round
    >
      <slot name="fab"></slot>
    </var-button>
  </div>
</template>

<script>
import VarButton from '../button'
import { defineComponent } from '../utils/create'
import { props } from './props'
import { isNumber } from '../utils/shared'
import { createParentMixin } from '../utils/mixins/relation'
import { doubleRaf } from '../utils/elements'

const RIGHT_HALF_SPACE_CLASS = 'var-bottom-navigation-item--right-half-space'
const LEFT_HALF_SPACE_CLASS = 'var-bottom-navigation-item--left-half-space'
const RIGHT_SPACE_CLASS = 'var-bottom-navigation-item--right-space'
const defaultFabProps = {
  type: 'primary',
}

export default defineComponent({
  name: 'VarBottomNavigation',
  components: { VarButton },

  mixins: [createParentMixin('bottomNavigation', { childrenKey: 'bottomNavigationItems' })],

  inheritAttrs: false,

  props,

  data: () => ({
    _fabProps: {},
  }),

  watch: {
    async bottomNavigationItems() {
      await doubleRaf()

      this.matchBoundary()
      this.removeMarginClass()

      if (!this.$slots.fab) {
        return
      }
      this.addMarginClass()
    },
    fabProps: {
      handler(newValue) {
        this._fabProps = { ...defaultFabProps, ...newValue }
      },
      immediate: true,
      deep: true,
    },
  },

  methods: {
    matchBoundary() {
      if (this.bottomNavigationItems.length === 0 || this.matchName() || this.matchIndex()) {
        return
      }

      this.handleActiveIndex()
    },
    matchName() {
      return this.bottomNavigationItems.find(({ name }) => {
        return this.active === name
      })
    },
    matchIndex() {
      return this.bottomNavigationItems.find(({ index }) => {
        return this.active === index
      })
    },
    handleActiveIndex() {
      if (!isNumber(this.active)) {
        return
      }

      const { getListeners } = this

      if (this.active < 0) {
        getListeners()['onUpdate:active']?.(0)
        this.itemsSetCurrent(0)
      } else if (this.active > this.bottomNavigationItems.length - 1) {
        getListeners()['onUpdate:active']?.(this.bottomNavigationItems.length - 1)
        this.itemsSetCurrent(this.bottomNavigationItems.length - 1)
      }
    },
    itemsSetCurrent(value) {
      this.bottomNavigationItems.forEach((item) => {
        item.setCurrent(value)
      })
    },
    onToggle(changedValue) {
      if (changedValue === this.active) {
        return
      }
      const { getListeners } = this

      getListeners().onBeforeChange ? this.handleBeforeChange(changedValue) : this.handleChange(changedValue)
    },
    handleBeforeChange(changedValue) {
      const { getListeners } = this
      Promise.resolve(getListeners().onBeforeChange?.(changedValue)).then(
        (res) => res && this.handleChange(changedValue)
      )
    },
    handleChange(changedValue) {
      const { getListeners } = this

      getListeners()['onUpdate:active']?.(changedValue)

      this.itemsSetCurrent(changedValue)

      getListeners().onChange?.(changedValue)
    },
    getBottomNavigationItems() {
      return Array.from(this.$refs.bottomNavigationDom.querySelectorAll('.var-bottom-navigation-item'))
    },
    removeMarginClass() {
      this.getBottomNavigationItems().forEach((dom) => {
        dom.classList.remove(RIGHT_HALF_SPACE_CLASS, LEFT_HALF_SPACE_CLASS, RIGHT_SPACE_CLASS)
      })
    },
    addMarginClass() {
      const isEven = this.bottomNavigationItems.length % 2 === 0

      this.getBottomNavigationItems().forEach((bottomNavigationItem, i) => {
        this.handleMarginClass(isEven, bottomNavigationItem, i, this.bottomNavigationItems.length)
      })
    },
    handleMarginClass(isEven, dom, i, length) {
      const isLast = i === length - 1
      if (!isEven && isLast) {
        dom.classList.add(RIGHT_SPACE_CLASS)
        return
      }

      const isFabLeft = i === length / 2 - 1
      const isFabRight = i === length / 2
      if (isFabLeft) {
        dom.classList.add(RIGHT_HALF_SPACE_CLASS)
      } else if (isFabRight) {
        dom.classList.add(LEFT_HALF_SPACE_CLASS)
      }
    },
    handleFabClick() {
      const { getListeners } = this
      getListeners().onFabClick?.()
    },
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../button/button';
@import './bottomNavigation';
</style>
