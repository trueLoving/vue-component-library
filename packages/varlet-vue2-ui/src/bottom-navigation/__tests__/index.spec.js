import example from '../example'
import BottomNavigation from '..'
import BottomNavigationItem from '../../bottom-navigation-item'
import VarBottomNavigation from '../BottomNavigation'
import VarBottomNavigationItem from '../../bottom-navigation-item/BottomNavigationItem'
import { mount } from '@vue/test-utils'
import Vue from 'vue'
import { delay } from '../../utils/jest'

test('test bottom-navigation example', () => {
  const wrapper = mount(example)

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test bottom-navigation plugin', () => {
  Vue.use(BottomNavigation).use(BottomNavigationItem)
  expect(Vue.component(BottomNavigation.name)).toBeTruthy()
})

test('test bottom-navigation before-change prevent switch', async () => {
  const onBeforeChange = jest.fn(() => false)
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    props: ['onBeforeChange'],
    data: () => ({
      active: 0,
    }),
    template: `
    <var-bottom-navigation :active.sync="active" @before-change="onBeforeChange">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper, {
    propsData: {
      onBeforeChange,
    },
  })
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item').at(1)
  await bottomNavigationItem.trigger('click')

  expect(onBeforeChange).toHaveBeenCalledTimes(1)
  expect(wrapper.vm.active).toBe(0)

  wrapper.destroy()
})

test('test bottom-navigation before-change return promise', async () => {
  const onBeforeChange = jest.fn(
    () =>
      new Promise((resolve) => {
        setTimeout(() => {
          resolve(true)
        }, 300)
      })
  )
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    props: ['onBeforeChange'],
    data: () => ({
      active: 0,
    }),
    template: `
    <var-bottom-navigation :active.sync="active" @before-change="onBeforeChange">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper, {
    propsData: {
      onBeforeChange,
    },
  })
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item').at(1)
  await bottomNavigationItem.trigger('click')
  expect(wrapper.vm.active).toBe(0)

  await delay(600)
  expect(onBeforeChange).toHaveBeenCalledTimes(1)
  expect(wrapper.vm.active).toBe(1)

  wrapper.destroy()
})

test('test bottom-navigation change event', async () => {
  let dummy
  const handleChange = jest.fn((active) => {
    dummy = active
  })
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    methods: {
      handleChange,
    },
    template: `
    <var-bottom-navigation :active.sync="active" @change="handleChange">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item').at(1)
  await bottomNavigationItem.trigger('click')

  expect(handleChange).toHaveBeenCalledTimes(1)
  expect(dummy).toBe(1)

  wrapper.destroy()
})

test('test bottom-navigation-item click event', async () => {
  let dummy
  const handleClick = jest.fn((active) => {
    dummy = active
  })
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    methods: {
      handleClick,
    },
    template: `
    <var-bottom-navigation :active.sync="active">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item @click="handleClick" label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(16)

  const bottomNavigationItem = wrapper.findAll('.var-bottom-navigation-item').at(1)
  await bottomNavigationItem.trigger('click')

  expect(handleClick).toHaveBeenCalledTimes(1)
  expect(dummy).toBe(1)

  wrapper.destroy()
})

test('test BottomNavigation relation BottomNavigationItems', async () => {
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    template: `
    <var-bottom-navigation :active.sync="active">
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
      <var-bottom-navigation-item label="标签" icon="account-circle" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(100)

  expect(wrapper.html()).toMatchSnapshot()
  await wrapper.setData({ active: 1 })

  await delay(50)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('fab button', async () => {
  const Wrapper = {
    components: {
      [VarBottomNavigation.name]: VarBottomNavigation,
      [VarBottomNavigationItem.name]: VarBottomNavigationItem,
    },
    data: () => ({
      active: 0,
    }),
    template: `
    <var-bottom-navigation :active.sync="active">
      <template #fab>
        <div class="tab_example" />
      </template>
      <var-bottom-navigation-item label="标签" icon="home" />
      <var-bottom-navigation-item label="标签" icon="magnify" />
      <var-bottom-navigation-item label="标签" icon="heart" />
    </var-bottom-navigation>
    `,
  }
  const wrapper = mount(Wrapper)
  await delay(100)

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})
