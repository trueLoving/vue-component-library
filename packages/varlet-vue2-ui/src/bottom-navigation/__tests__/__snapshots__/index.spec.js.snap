// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fab button 1`] = `
"<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"> <button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item var-bottom-navigation-item--right-space\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-button var--box var-bottom-navigation__fab var-button--normal var--inline-flex var-button--primary var-elevation--2 var-button--round var-bottom-navigation--fab-right\\">
    <!---->
    <div class=\\"var-button__content\\">
      <div class=\\"tab_example\\"></div>
    </div>
  </button></div>"
`;

exports[`test BottomNavigation relation BottomNavigationItems 1`] = `
"<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
  <!---->
</div>"
`;

exports[`test BottomNavigation relation BottomNavigationItems 2`] = `
"<div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
    <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
  <!---->
</div>"
`;

exports[`test bottom-navigation example 1`] = `
"<div class=\\"bottom-navigation-example\\">
  <div class=\\"app-type\\">基本使用</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!---->
  </div>
  <div class=\\"app-type\\">通过名称匹配</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!---->
  </div>
  <div class=\\"app-type\\">徽标提示</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <div class=\\"var-badge var--box var-bottom-navigation-item__badge\\">
        <transition-stub name=\\"var-badge-fade\\"><span class=\\"var-badge__content var-badge--danger var-badge__dot var-badge__dot--right\\"><span></span></span></transition-stub>
      </div> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span>
    </button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <div class=\\"var-badge var--box var-bottom-navigation-item__badge\\">
        <transition-stub name=\\"var-badge-fade\\"><span class=\\"var-badge__content var-badge--primary\\"><span>66</span></span></transition-stub>
      </div> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span>
    </button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!---->
  </div>
  <div class=\\"app-type\\">自定义颜色</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!---->
  </div>
  <div class=\\"app-type\\">监听切换事件</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!---->
  </div>
  <div class=\\"app-type\\">监听点击事件</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1;\\"><button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-account-circle\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!---->
  </div>
  <div class=\\"app-type\\">悬浮按钮</div>
  <div class=\\"var-bottom-navigation\\" style=\\"z-index: 1; margin-top: 10px;\\"> <button class=\\"var-bottom-navigation-item var-bottom-navigation-item--active\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-home\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button> <button class=\\"var-bottom-navigation-item\\"><i class=\\"var-icon var-bottom-navigation-item__icon var-icon--set var-icon-bell\\" style=\\"transition: transform 0ms;\\"></i>
      <!----> <span class=\\"var-bottom-navigation-item__label\\">
      标签
     </span></button>
    <!----> <button class=\\"var-button var--box var-bottom-navigation__fab var-button--normal var--inline-flex var-button--primary var-elevation--2 var-button--round var-bottom-navigation--fab-center\\">
      <!---->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-heart\\" style=\\"transition: transform 0ms;\\"></i></div>
    </button></div>
</div>"
`;
